src/main/java/com/elastic/support/BaseInputs.java

```diff
@@ -2,57 +2,147 @@
+
     }
 
-    @Parameter(names = {"-?", "--help"}, description = "Help contents.", help = true)
-    protected boolean help;
-    public boolean isHelp() {
-        return help;
+    public void usage(){
+        if(jCommander != null){
+            jCommander.usage();
+        }
+        else{
+            logger.info("Please rerun with the --help option or with no arguments for interactive mode.");
+        }
     }
-    public void setHelp(boolean help) {
-        this.help = help;
+
+    protected void runOutputDirInteractive(){
+        outputDir = ResourceCache.textIO.newStringInputReader()
+                .withValueChecker(( String val, String propname) -> validateFile(val))
+                .read(SystemProperties.lineSeparator + outputDirDescription);
     }
 
-    @Parameter(names = {"-o", "--out", "--output", "--outputDir"}, description = "Fully qualified path to output directory.")
-    // If no output directory was specified default to the working directory
-    protected String outputDir = SystemProperties.userDir;
-    public String getOutputDir() {
-        return outputDir;
+    public List<String> validatePort(int val){
+        if (val < 1 || val > 65535){
+            return Collections.singletonList("Outside the valid range of port values. 1-65535 ");
+        }
+        return null;
     }
-    public void setOutputDir(String outputDir) {
-        this.outputDir = outputDir;
+
+    public List<String> validateFile(String val) {
+        if (StringUtils.isEmpty(val.trim())) {
+            return null;
+        }
+
+        File file = new File(val);
+
+        if (!file.exists()) {
+            return Collections.singletonList("Specified file could not be located.");
+        }
+
+        return null;
+
     }
 
+    public List<String> validateDir(String val) {
+        if (StringUtils.isEmpty(val.trim())) {
+            return null;
+        }
 
-    public boolean validate() {
+        File file = new File(val);
 
-        // If we're in help just shut down.
-        if (isHelp()) {
-            this.jCommander.usage();
-            return false;
+        if (!file.exists() && file.isDirectory()) {
+            return Collections.singletonList("Specified directory location could not be located.");
         }
-        return true;
+
+        return null;
+
+    }
+
+    public List<String> validateRequiredFile(String val) {
+        if (StringUtils.isEmpty(val.trim())) {
+            return Collections.singletonList("Input file is required.");
+        }
+
+        File file = new File(val);
+
+        if (!file.exists() && file.isDirectory()) {
+            return Collections.singletonList("Specified required file location could not be located.");
+        }
+
+        return null;
+
     }
 
     public String toString(){
-        return "Output Directory: " + outputDir + SystemProperties.lineSeparator;
+       return "outputDir: " + outputDir + ",";
```

Comment:
Seems like it should be the job of the caller to add the `","` if it wants it.

Reply:
That one is kind of trivial, but this is just the base class. It's never used by itself. I did go back and check the diagnostic inputs and when I regenerated the toString I lost the call the the superclass. So I'm fixing the DiagnosticInputs as well.